frontend/src/views/admin/matching/applicant-view/grid/grid-item/dropdown.tsx

```diff
@@ -107,28 +75,62 @@ export function GridItemDropdown({
                     <Dropdown.Item onClick={() => setShowChangeHours(true)}>
                         Change assigned hours
                     </Dropdown.Item>
-                    <Dropdown.Item onClick={unassignFromPosition}>
+                    <Dropdown.Item
+                        onClick={UnassignApplicantFromPosition(
+                            baseMatchValues.positionCode,
+                            baseMatchValues.utorid
+                        )}
+                    >
                         Unassign from <b>{match.position.position_code}</b>
                     </Dropdown.Item>
                 </>
             )}
+            {canBeStarred && (
+                <Dropdown.Item onClick={SetStarred(match, true)}>
```

Comment:
This is very weird. You should have a `toggleStar` action instead. This should also be put in its own component, so that starring can be worried about in a single place.